mtb-hal-cat1/CMakeLists.txt

```diff
@@ -0,0 +1,69 @@
+zephyr_library_sources_ifdef(CONFIG_SOC_PACKAGE_PSOC6_03_68_QFN COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/pin_packages/cyhal_psoc6_03_68_qfn.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_PACKAGE_PSOC6_04_64_TQFP COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/pin_packages/cyhal_psoc6_04_64_tqfp.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_PACKAGE_PSOC6_04_68_QFN COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/pin_packages/cyhal_psoc6_04_68_qfn.c)
+zephyr_library_sources_ifdef(CONFIG_SSOC_PACKAGE_PSOC6_04_80_TQFP COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/pin_packages/cyhal_psoc6_04_80_tqfp.c)
+
+zephyr_library_sources_ifdef(CONFIG_SOC_DIE_PSOC6_01 COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/triggers/cyhal_triggers_psoc6_01.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_DIE_PSOC6_02 COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/triggers/cyhal_triggers_psoc6_02.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_DIE_PSOC6_03 COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/triggers/cyhal_triggers_psoc6_03.c)
+zephyr_library_sources_ifdef(CONFIG_SOC_DIE_PSOC6_04 COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/triggers/cyhal_triggers_psoc6_04.c)
+
+# High level interface for interacting with CAT1 hardware
+zephyr_library_sources_ifdef(CONFIG_SERIAL COMPONENT_PSOC6HAL/source/cyhal_uart.c)
+zephyr_library_sources_ifdef(CONFIG_ADC COMPONENT_PSOC6HAL/source/cyhal_adc_sar.c)
+zephyr_library_sources_ifdef(CONFIG_I2C COMPONENT_PSOC6HAL/source/cyhal_i2c.c)
+zephyr_library_sources_ifdef(CONFIG_SPI COMPONENT_PSOC6HAL/source/cyhal_spi.c)
+zephyr_library_sources_ifdef(CONFIG_COUNTER COMPONENT_PSOC6HAL/source/cyhal_timer.c)
+zephyr_library_sources_ifdef(CONFIG_PWM COMPONENT_PSOC6HAL/source/cyhal_pwm.c)
+zephyr_library_sources_ifdef(CONFIG_WATCHDOG COMPONENT_PSOC6HAL/source/cyhal_wdt.c)
+zephyr_library_sources_ifdef(CONFIG_LPTIMER_INFINEON_CAT1 COMPONENT_PSOC6HAL/source/cyhal_lptimer.c)
+zephyr_library_sources_ifdef(CONFIG_RTC_INFINEON_CAT1 COMPONENT_PSOC6HAL/source/cyhal_rtc.c)
+zephyr_library_sources_ifdef(CONFIG_TRNG_INFINEON_CAT1 COMPONENT_PSOC6HAL/source/cyhal_trng.c)
+
+if(CONFIG_ADC)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_analog_common.c)
+endif()
+
+if(CONFIG_COUNTER)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_tcpwm_common.c)
+endif()
+
+if(CONFIG_SERIAL OR CONFIG_I2C OR CONFIG_SPI)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_scb_common.c)
+endif()
+
+# Common part
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_clock.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_gpio.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_hwmgr.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_interconnect.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_sdhc.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_syspm.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_system.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_utils.c)
+zephyr_library_sources(COMPONENT_PSOC6HAL/source/cyhal_utils_psoc.c)
```

Comment:
This uses some Zephyr Kconfig to enable code at hal_infineon. Problem is,majority of it doesn't have a driver at Zephyr side and maintainers wont be aware of this.

I don't like this approach because any changes on Zephyr Kconfigs can result in broken platform.

I think a better way is follow how other vendors are doing, ex. [GD32](https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/hal_gigadevice/CMakeLists.txt). In this example vendor define `CONFIG_USE_<VENDOR>_<PERIPHERAL>` which includes all necessary dependencies. It is defined at `hal_<VENDOR>` and controlled at driver Kconfig. In case of serial driver, for instance, [Serial Kconfig.gd32](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/Kconfig.gd32)